packages/kit/src/core/env.js

```diff
@@ -23,6 +24,14 @@ export function create_module(id, env) {
 	return GENERATED_COMMENT + declarations.join('\n\n');
 }
 
+/**
+ * @param {boolean} pub
+ * @returns {string}
+ */
+export function create_dynamic_module(pub) {
+	return `export { env } from '${runtime_base}/env-${pub ? 'public' : 'private'}.js';`;
+}
```

Comment:
Subjective, but I think this is more immediately grokkable

```suggestion
/**
 * @param {'public' | 'private'} type
 * @returns {string}
 */
export function create_dynamic_module(type) {
	return `export { env } from '${runtime_base}/env-${type}.js';`;
}
```